api/pkg/apps/cms/types.go

```diff
@@ -61,6 +79,29 @@ type CommentList struct {
 type CaseTemplate struct {
 	// FormElements is an ordered list of the elements found in the form
 	FormElements []CaseTemplateFormElement `json:"formElements" bson:"formElements"`
+
+	// FormData holds actual data corresponding to the elements in FormElements
+	FormData CaseFormData `json:"formData" bson:"formData"`
+}
+
+// NewCaseTemplate constructs a new case template from a JSON template string and creates a CaseFormData data structure
+// that with hold the data corresponding to the fields described in the template
+func NewCaseTemplate(templateString string) (*CaseTemplate, error) {
+	var ct CaseTemplate
+	if err := json.Unmarshal([]byte(templateString), &ct); err != nil {
+		return &ct, err
+	}
+	ct.FormData = ct.ToCaseFormData()
+	return &ct, nil
+}
```

Comment:
This method should not be needed. We can simply use the vanilla `json.Unmarshal/json.Marshal`
The `values` of a `CaseTemplate` shoudl be stored in the `caseTemplate...Attribute.Value`